setup.sh

```
@@ -0,0 +1,129 @@
+                if [[ ${1:0:2} == "--" ]]; then
+                    # Handle long options
+                    case $1 in
+                        --fresh) FRESH_INSTALL=true ;;
+                        --dependencies) INSTALL_DEPS=true ;;
+                        --rebuild) REBUILD_APP=true ;;
+                        --migrate) MIGRATE_DB=true ;;
+                        --cache) CLEAR_CACHE=true ;;
+                        --help) show_help ;;
+                        *) echo -e "${RED}Unknown parameter: $1${NO_COLOR}"; exit 1 ;;
+                    esac
+                    shift
+                    break
+                else
+                    # Handle combined short options
+                    for (( i=1; i<${#1}; i++ )); do
+                        case ${1:$i:1} in
+                            f) FRESH_INSTALL=true ;;
+                            d) INSTALL_DEPS=true ;;
+                            r) REBUILD_APP=true ;;
+                            m) MIGRATE_DB=true ;;
+                            c) CLEAR_CACHE=true ;;
+                            h) show_help ;;
+                            *) echo -e "${RED}Unknown parameter: -${1:$i:1}${NO_COLOR}"; exit 1 ;;
+                        esac
+                    done
+                    shift
+                    break
+                fi
+            done
+            ;;
+        *)
+            echo -e "${RED}Unknown parameter: $1${NO_COLOR}"
+            exit 1
+            ;;
+    esac
+done
+
+if [[ "$FRESH_INSTALL" == true ]]; then
+    INSTALL_DEPS=true
+    REBUILD_APP=true
+    MIGRATE_DB=true
+    CLEAR_CACHE=true
+fi
+
+if [[ "$INSTALL_DEPS" == true ]]; then
+    echo -e "${BLUE}Installing Laravel dependencies inside of a container...${NO_COLOR}"
+    docker run --rm \
+        -u "$(id -u):$(id -g)" \
+        -v "$(pwd)":/var/www/html \
+        -w /var/www/html \
+        laravelsail/php82-composer:latest \
+        composer install --optimize-autoloader
+fi
+
```

Ideally I would like to have some level of error handling here just in case teh user runs the script without doing the `-f/--fresh` option first. If the `vendor` directory can't be found and the `-f/--fresh` option is not true, then the application should throw an error like
```bash
if [[ ! -d 'vendor/' && "$FRESH_INSTALL" != true ]]; then
    echo -e "${RED}Sail was not found, run './setup.sh --fresh' to build it.${NO_COLOR}" && exit 1
fi
```
Although you might be able to omit the ` && "$FRESH_INSTALL" != true` part as the install dependencies runs before this.

---

Added a check to ensure `vendor/` exists before any `sail` commands are run. Also added a preflight check to make sure Docker is installed and running.

Script has also changed a bit in other ways since your last review.

Now, the only flags are `--fresh` and `--help`. If you don't use `--fresh`, there are interactive prompts to determine which actions to take. Makes it a little nicer to use and gives more fine-grained control.

Also did some general cleanup and put everything into functions to make it easier to read.